cmd/admin-handlers.go

```diff
@@ -1199,15 +1200,51 @@ func (a adminAPIHandlers) OBDInfoHandler(w http.ResponseWriter, r *http.Request)
 		finish()
 	}
 
+	deadlineSeconds := time.Duration(3600)
+	deadline := r.URL.Query().Get("deadlineseconds")
+	if deadline != "" {
+		deadlineSecondsInt, err := strconv.Atoi(deadline)
+		if err != nil {
+			errResp(err)
+			return
+		}
+		deadlineSeconds = time.Duration(deadlineSecondsInt)
```

Comment:
perhaps [time.ParseDuration](https://golang.org/pkg/time/#ParseDuration) would be more idiomatic